bundler/doc/development/DEBUGGING.md

```diff
@@ -0,0 +1,76 @@
+# Debugging
+
+Debugging Bundler can be challenging, don't be discouraged 🤗.
+
+Make sure you've followed the [development setup](SETUP.md) docs before trying to debug.
+
+## Debugging in tests
+
+Your best option is to print debug in the test suite. Put `puts` statements anywhere you want to see an object or variable and you'll see your `puts` in the console output.
+
+```ruby
+puts "@definition: #{@definition}"
+puts "specification.class.name: #{specification.class.name}"
+puts "spec.method(:to_checksum).source_location: #{spec.method(:to_checksum).source_location}"
+# etc
+```
+
+Most tests use Open3 to run Bundler in a sub process and capture the output into a string, which makes it impossible to use pry even if you can get it to load.
+
+## Interactive debugging locally
+
+When testing Bundler locally, you can use pry for interactive debugging.
+
+The easiest way to test locally is to set up a directory with a Gemfile and run your Bundler shell alias (see the [development setup](SETUP.md) docs for instructions on setting up the alias).
+
+We recommend putting this directory inside of `tmp` so that your local tests don't accidentally get committed.
+
+```bash
+cd tmp
+mkdir [name of directory for local testing] && cd [name of directory for local testing]
+dbundle init
```

Comment:
I normally point directly to my working copy with something like `/path/to/rubygems/bundle/exe/bundle`. These days once you land on a copy of Bundler, you never get out of it because we always use `require_relative`, so messing with the `$LOAD_PATH` should not be necessary.

Reply:
Ah so should we change the alias instructions in the setup doc?

Reply:
I don't mind the current recommendation to set an alias, and it should also work just fine, I don't think you need to change anything. Just wanted to mention the way I do it :)

Reply:
I think everyone have different debugging habits :). I was using `ruby -Ilib -s ./exe/bundle -- --version` and even today it is usually enough to just call `./exe/bundle`. 

But indeed, `dbundle` is mentioned at various places. It is ok to keep it in here as well.